cmd/deploy.go

```diff
@@ -95,192 +88,312 @@ func postDeploymentRequest(l log.Logger, projectName string, namespaceName strin
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Error(coloredError("Deployment process took too long, timing out"))
+		}
+		return errors.New("deployement failed")
+	}
+	var specFound bool
+	for _, namespaceName := range selectedNamespaceNames {
+		namespace, err := conf.GetNamespaceByName(namespaceName)
+		if err != nil {
+			return err
+		}
+		jobSpecFs := afero.NewBasePathFs(afero.NewOsFs(), namespace.Job.Path)
+		jobSpecRepo := local.NewJobSpecRepository(
+			jobSpecFs,
+			local.NewJobSpecAdapter(pluginRepo),
+		)
+		l.Info(fmt.Sprintf("\n> [%s] Deploying jobs", namespaceName))
+		jobSpecs, err := jobSpecRepo.GetAll()
+		if err != nil {
+			return err
+		}
+		if len(jobSpecs) == 0 {
+			fmt.Printf("[%s] skipping as job spec is empty\n", namespaceName)
+			continue
+		}
+
+		var adaptedJobSpecs []*pb.JobSpecification
+		adapt := v1handler.NewAdapter(pluginRepo, datastoreRepo)
+		for _, spec := range jobSpecs {
+			adaptJob, err := adapt.ToJobProto(spec)
+			if err != nil {
+				return fmt.Errorf("[%s] failed to serialize: %s - %w", namespaceName, spec.Name, err)
+			}
+			adaptedJobSpecs = append(adaptedJobSpecs, adaptJob)
+		}
+		specFound = true
+		if err := stream.Send(&pb.DeployJobSpecificationRequest{
+			Jobs:          adaptedJobSpecs,
+			ProjectName:   conf.Project.Name,
+			NamespaceName: namespaceName,
+		}); err != nil {
+			return fmt.Errorf("[%s] deployment failed: %w", namespaceName, err)
+		}
 	}
-	if err = registerNamespace(deployTimeoutCtx, l, runtime, projectSpec.Name, namespaceSpec); err != nil {
+	if err := stream.CloseSend(); err != nil {
 		return err
 	}
+	if !specFound {
+		return nil
+	}
 
-	if !ignoreResources {
-		// deploy datastore resources
-		for storeName, repoFS := range datastoreSpecFs {
-			l.Info(fmt.Sprintf("\n> Deploying resources for %s", storeName))
+	var counter int
+	var streamErrs []error
+	spinner := NewProgressBar()
+	if !verbose {
+		spinner.StartProgress(len(selectedNamespaceNames), "please wait")
+	}
+	for {
+		resp, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		if resp.GetAck() {
+			if !resp.GetSuccess() {
+				streamErrs = append(streamErrs, errors.New(resp.GetMessage()))
+			}
+			counter++
+			spinner.SetProgress(counter)
+		}
+	}
+	spinner.Stop()
+	if len(streamErrs) > 0 {
+		for _, e := range streamErrs {
+			fmt.Println(e)
+		}
+		return errors.New("one or more errors are encountered during deploy")
+	}
+	return nil
+}
+
+func deployAllResources(deployTimeoutCtx context.Context,
+	runtime pb.RuntimeServiceClient,
+	l log.Logger, conf config.Optimus, pluginRepo models.PluginRepository,
+	datastoreRepo models.DatastoreRepo,
+	datastoreSpecFs map[string]map[string]afero.Fs,
+	namespaceNames []string,
+	verbose bool,
+) error {
+	var selectedNamespaceNames []string
+	if len(namespaceNames) > 0 {
+		selectedNamespaceNames = namespaceNames
+	} else {
+		for _, namespace := range conf.Namespaces {
```

Comment:
Can use spread for values of append, eg
```golang
selectedNamespaceNames = append(selectedNamespaceNames, ...conf.Namespaces)
```

Can be part of separate pr.

Reply:
What we need is the slice of namespace names, not the namespaces themselves. That's why we iterate it using a loop.